content/master/concepts/compositions.md

```diff
@@ -1308,47 +1308,69 @@ parameters in the Composition.
 * Composition function `type` is `container`.
 * Composition function names are unique.
 
-### Resource schema validation
-<!-- vale write-good.TooWordy = NO -->
-Optionally, Crossplane can also validate the schema of the resources defined
-inside a Composition. This verifies that the resource `apiVersion` and `kinds`
-are valid. 
-<!-- vale write-good.TooWordy = YES -->
+### Composition schema aware validation
 
-Enable "schema validation" with the
-`--enable-composition-webhook-schema-validation` flag on the Crossplane pod.
+On top of the preceding integrity checks, Crossplane performs schema aware
+validation of Compositions, checking that `patches`, `readinessChecks` and
+`connectionDetails` are valid according to the involved resources' schemas, for
+example checking that the source/destination field of a patch is a valid field
+according to the source/destination resource's schema and that types match, if
+defined, taking into account transforms.
+
+{{<hint "note" >}}
+Composition schema aware validation is a beta feature, so Crossplane
+enables it by default. You can disable it by setting the
+`--enable-composition-webhook-schema-validation=false` flag on the Crossplane
+pod.
 The [Crossplane Pods]({{<ref "./pods#edit-the-deployment">}}) page has
 more information on enabling Crossplane flags.
+{{< /hint >}}
+
+#### Schema aware validation modes
+
+Integrity validation errors always result in a rejection of the Composition.
+
+You can configure how Crossplane should handle both missing involved resources'
+schemas and actual schema aware validation errors.
```

Comment:
```suggestion
<!-- vale alex.ProfanityUnlikely = NO -->
Integrity validation errors reject a Composition.
<!-- vale alex.ProfanityUnlikely = Yes  -->

Set the schema validation mode to configure how Crossplane handles both missing 
resource schemas and schema validation errors.
```

The word "reject" is a vale error and shouldn't be but I can't make the change in review so I just disable the check.

Reply:
I get the intention, but I feel it's slightly wrong saying that errors reject the composition... it's crossplane rejecting it because of errors. I'd rather be correct than have vale somewhat happy 😬

Reply:
I'm trying to get away from "result in a rejection". How about 

"Crossplane rejects Compositions with validation errors"

?

The vale error from an imported library that says the word "reject" may be profane, which it isn't in our writing/use.

Reply:
```suggestion
Crossplane always rejects Compositions in case of integrity errors.

Set the schema validation mode to configure how Crossplane handles both missing 
resource schemas and schema aware validation errors.
```

Reply:
LGTM